src/praisonai-bot/praisonai_bot/bots/bot.py

```
@@ -272,20 +287,88 @@ def _build_adapter(self) -> Any:
 
         self._adapter = self._build_adapter()
         self._is_running = True
-        await self._adapter.start()
+
+        if not self._supervision_enabled():
+            await self._adapter.start()
+            return
+
+        try:
+            from ..gateway.supervisor import ChannelSupervisor
+        except Exception as exc:  # pragma: no cover - optional supervision layer
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)